target_chains/ethereum/entropy_sdk/solidity/MockEntropy.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+interface IEntropyConsumer {
+    function _entropyCallback(
+        uint64 sequenceNumber,
+        address provider,
+        bytes32 randomNumber
+    ) external;
+}
+
+contract MockEntropy {
+    uint64 public sequenceNumber;
+    uint128 public constant FEE = 0.000015 ether;
+
+    function getDefaultProvider() external view returns (address) {
+        return address(this);
+    }
+
+    function getFee(address provider) external pure returns (uint128) {
+        require(provider != address(0), "Invalid provider address");
+        return FEE;
+    }
+
+    function requestWithCallback(
+        address provider,
+        bytes32 userRandomNumber
+    ) external payable returns (uint64) {
+        require(provider != address(0), "Invalid provider address");
+        require(msg.value >= FEE, "Not enough ether sent for fee");
+        return sequenceNumber++;
+    }
+
+    function triggerCallback(
+        uint64 _sequenceNumber,
+        uint256 _randomNumber,
+        address _callbackAddress
+    ) external {
+        bytes32 randomNumberBytes = bytes32(_randomNumber);
```

Comment:
It should not be possible to trigger the callback for the same sequence number twice. The callback address should come from the sequence number (stored in the contract) and not as another parameter

Reply:
Added a better way to manage callbacks.

Users can now make as many request as they would like. Upon calling triggerCallback() the request will happen in order of the request made. If there are no pending request, triggerCallback will revert.

Reply:
Realized I misread this. I will store the contract address that created the request and update.